RestEasy Jackson test fails in certain time zones

### Describe the bug

In the `America/Chicago` time zone at `2024-03-16` at time `10:59:00` the test `io.quarkus.it.resteasy.jackson.GreetingResourceTest#testEndpoint` fails with an exception like this:

```
[ERROR] Tests run: 3, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.108 s <<< FAILURE! -- in io.quarkus.it.resteasy.jackson.GreetingResourceTest
[ERROR] io.quarkus.it.resteasy.jackson.GreetingResourceTest.testEndpoint -- Time elapsed: 0.081 s <<< FAILURE!
java.lang.AssertionError: 
1 expectation failed.
JSON path sqlDate doesn't match.
Expected: 2019-01-01
  Actual: 2018-12-31
```

It looks like a time zone handling problem (hence the zone info above). Notable is that this failure was observed during the period when the US is on DST but other countries are not (this may or may not be a factor but should be considered).

See also #37659 which is likely a related problem.

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

Darwin xxxx 23.3.0 Darwin Kernel Version 23.3.0: Wed Dec 20 21:31:00 PST 2023; root:xnu-10002.81.5~7/RELEASE_ARM64_T6020 arm64

### Output of `java -version`

openjdk version "17.0.9" 2023-10-17 OpenJDK Runtime Environment Temurin-17.0.9+9 (build 17.0.9+9) OpenJDK 64-Bit Server VM Temurin-17.0.9+9 (build 17.0.9+9, mixed mode)

### Quarkus version or git rev

ffeea8f53f9a59145a073b0bd26b1791f156835e

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.5 (57804ffe001d7215b5e7bcb531cf83df38f93546) Maven home: /Users/david/local/apache-maven Java version: 17.0.9, vendor: Eclipse Adoptium, runtime: /Library/Java/JavaVirtualMachines/temurin-17.jdk/Contents/Home Default locale: en_US, platform encoding: UTF-8 OS name: "mac os x", version: "14.3.1", arch: "aarch64", family: "mac"

### Additional information

_No response_

--- a/integration-tests/resteasy-jackson/src/test/java/io/quarkus/it/resteasy/jackson/GreetingResourceTest.java
+++ b/integration-tests/resteasy-jackson/src/test/java/io/quarkus/it/resteasy/jackson/GreetingResourceTest.java
@@ -6,7 +6,6 @@ import static org.hamcrest.CoreMatchers.equalTo;
 
 import java.sql.Date;
 import java.time.LocalDate;
-import java.time.ZoneOffset;
 
 import javax.xml.datatype.DatatypeConfigurationException;
 import javax.xml.datatype.DatatypeFactory;
@@ -24,7 +23,7 @@ class GreetingResourceTest {
     void testEndpoint() throws DatatypeConfigurationException {
 
         final LocalDate localDate = LocalDate.of(2019, 01, 01);
-        final Date sqlDate = new Date(localDate.atStartOfDay(ZoneOffset.UTC).toInstant().toEpochMilli());
+        final Date sqlDate = new Date(119, 0, 1);
         final XMLGregorianCalendar xmlGregorianCalendar = DatatypeFactory.newInstance()
                 .newXMLGregorianCalendar("2019-01-01T00:00:00.000+00:00");
         final Greeting greeting = new Greeting("hello", localDate, sqlDate, xmlGregorianCalendar);

